src/main/java/org/sagebionetworks/bridge/validators/AppConfigValidator.java

```diff
@@ -205,16 +206,26 @@ public void validate(Object object, Errors errors) {
             for (int i=0; i < appConfig.getAssessmentReferences().size(); i++) {
                 AssessmentReference ref = appConfig.getAssessmentReferences().get(i);
                 errors.pushNestedPath("assessmentReferences["+i+"]");
+                boolean validAppId = true;
+                if (ref.getAppId() == null) {
+                    validAppId = false;
+                    errors.rejectValue("appId", "is required");
+                } else if (!ref.getAppId().equals(appConfig.getAppId()) && !ref.getAppId().equals(SHARED_APP_ID)) {
+                    validAppId = false;
+                    errors.rejectValue("appId", "does not refer to a valid app");
+                }
                 if (ref.getGuid() == null) {
                     errors.rejectValue("guid", "is required");
                 } else {
                     if (uniqueRefs.contains(ref)) {
                         errors.rejectValue("guid", "refers to the same assessment as another reference");
                     } else {
-                        try {
-                            assessmentService.getAssessmentByGuid(appConfig.getAppId(), null, ref.getGuid());
-                        } catch(EntityNotFoundException e) {
-                            errors.rejectValue("guid", "does not refer to an assessment");
+                        if (validAppId) {
+                            try {
+                                assessmentService.getAssessmentByGuid(ref.getAppId(), null, ref.getGuid());
+                            } catch(EntityNotFoundException e) {
+                                errors.rejectValue("guid", "does not refer to an assessment in given app");
+                            }
```

Comment:
Can this be moved out of all these if/then blocks to be a check at the same level as checking appId and guid? We may need to check that appId and guid are not null, but then the test is basically separate from these other tests.